### PR TITLE
add support for @Blocking and @NonBlocking

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -16,6 +16,45 @@ SmallRye Fault Tolerance is an implementation of {microprofile-fault-tolerance}[
...
+We also want to stay compatible with the MicroProfile Fault Tolerance specification as much as possible.
+For these reasons, SmallRye Fault Tolerance only considers these annotations for methods that use some fault tolerance strategy.
+
+===== Recommendation
+
+For methods that use fault tolerance and return `CompletionStage`, we recommend to declare their `@Blocking` or `@NonBlocking` nature.
+In such case, the `@Asynchronous` annotation becomes optional.
+
+We also recommend to avoid `@Asynchronous` methods that return `Future`, because the only way to obtain the future value is blocking.
+
```
Comment (Ladicek) [inline README.adoc:57]: Hi @cescoffier, could you please review this documentation piece? I'd like to know if this usage of `@Blocking`/`@NonBlocking` aligns with your usage and your expectations around these annotations. If the documentation misses something or is unclear, pointing that out would be great too :-) Thank you!

File: README.adoc [hunk 2]
```diff
@@ -16,6 +16,45 @@ SmallRye Fault Tolerance is an implementation of {microprofile-fault-tolerance}[
...
+These annotations can have multiple meanings, depending on context.
+SmallRye Fault Tolerance only pays attention to these annotations if:
+
+* they are placed on methods that return `CompletionStage` (the `Future` type can't really be used for non-blocking processing);
+* they are placed on methods that apply some fault tolerance strategy (such as `@Fallback`, defined either on a method or a class).
+
+Under these circumstances, SmallRye Fault Tolerance assigns these annotations the following meaning:
+
+* `@Blocking` means that execution of the operation will be offloaded to another thread (that's what `@Asynchronous` does by default);
```
Comment (cescoffier) [inline README.adoc:34]: I would add something like: `Uses this annotation if you don't want to block the caller thread and the annotated method has blocking logic.`

**Review by cescoffier — APPROVED** [review]
It looks good to me. I've done a small suggestion, but I already more or less covered in the _recommendation_ section.

At some point, we would need to add `Uni` support ;-)

**Comment by Ladicek** [comment]
OK, thank you, I'll incorporate that suggestion. There's #219 for tracking the `Uni` support. I have more time to work on SRye FT now, so it should happen in the coming months :-)

**Pull request merged**